### PR TITLE
Implement 0x4c89ae

### PR DESCRIPTION
**Pull request by petergaal**
- implement handling horizontal scrollbar button right press

### CODE REVIEW

**Comment by AaronVanGeffen** [comment]
This function has also been implemented in #348.

File: src/openloco/input/mouse_input.cpp [hunk 1]
```diff
@@ -1426,8 +1426,23 @@ namespace openloco::input
                 break;
             }
             case ui::scrollview::scroll_part::hscrollbar_button_right:
-                call(0x4c89ae, regs);
+            {
+                ui::window* w = WindowManager::find(_pressedWindowType, _pressedWindowNumber);
+                if (w != nullptr)
+                {
+                    w->scroll_areas[scrollAreaIndex].flags |= scroll_flags::HSCROLLBAR_RIGHT_PRESSED;
+                    int16_t widget_width = w->widgets[_pressedWidgetIndex].right - w->widgets[_pressedWidgetIndex].left - 1;
```
Comment (duncanspumpkin) [inline src/openloco/input/mouse_input.cpp:1434]: variables should be `camelCase`. Also use the function `w->widgets[_pressedWidgetIndex].width()`

File: src/openloco/input/mouse_input.cpp [hunk 2]
```diff
@@ -1426,8 +1426,23 @@ namespace openloco::input
...
+                    if ((w->scroll_areas[scrollAreaIndex].flags & scroll_flags::VSCROLLBAR_VISIBLE) != 0)
```
Comment (duncanspumpkin) [inline src/openloco/input/mouse_input.cpp:1435]: In general we dont do `!= 0` as its implicit

File: src/openloco/input/mouse_input.cpp [hunk 3]
```diff
@@ -1426,8 +1426,23 @@ namespace openloco::input
...
+                    {
+                        widget_width -= SCROLLBAR_WIDTH + 1;
+                    }
+                    int16_t widget_content_width = std::max(w->scroll_areas[scrollAreaIndex].h_right - widget_width, 0);
+                    w->scroll_areas[scrollAreaIndex].h_left = std::min((int16_t)(w->scroll_areas[scrollAreaIndex].h_left + 3), widget_content_width);
```
Comment (duncanspumpkin) [inline src/openloco/input/mouse_input.cpp:1440]: prefer static casts over c style casts. I think you can also do `std::min<int16_t>(w->scroll_areas[scrollAreaIndex].h_left + 3, widget_content_width);`

File: src/openloco/input/mouse_input.cpp [hunk 4]
```diff
@@ -1426,8 +1426,23 @@ namespace openloco::input
...
+                ui::window* w = WindowManager::find(_pressedWindowType, _pressedWindowNumber);
+                if (w != nullptr)
+                {
+                    w->scroll_areas[scrollAreaIndex].flags |= scroll_flags::HSCROLLBAR_RIGHT_PRESSED;
+                    int16_t widget_width = w->widgets[_pressedWidgetIndex].right - w->widgets[_pressedWidgetIndex].left - 1;
+                    if ((w->scroll_areas[scrollAreaIndex].flags & scroll_flags::VSCROLLBAR_VISIBLE) != 0)
+                    {
+                        widget_width -= SCROLLBAR_WIDTH + 1;
+                    }
+                    int16_t widget_content_width = std::max(w->scroll_areas[scrollAreaIndex].h_right - widget_width, 0);
```
Comment (duncanspumpkin) [inline src/openloco/input/mouse_input.cpp:1439]: `camelCase`

File: src/openloco/input/mouse_input.cpp (hunk 2 above)
Comment (petergaal) [reply src/openloco/input/mouse_input.cpp:1435]: I think we should write every time as this avoids confusion. Not putting the boolean expression is error prone, considered as a bad practice (from old C times) and in other languages it's even not possible to do.

**Pull request merged**